pwiz_tools/Skyline/SkylineGraphs.cs

```diff
@@ -1104,14 +1104,21 @@ void GraphSpectrum.IStateProvider.BuildSpectrumMenu(bool isProteomic, ZedGraphCo
             menuStrip.Items.Insert(iInsert++, lockYaxisContextMenuItem);
             menuStrip.Items.Insert(iInsert++, toolStripSeparator14);
 
-            // Need to test small mol
-            if (isProteomic && control?.ControlType == SpectrumControlType.LibraryMatch)
-            {
-                prositLibMatchItem.Checked = Settings.Default.Prosit;
-                menuStrip.Items.Insert(iInsert++, prositLibMatchItem);
-                mirrorMenuItem.Checked = Settings.Default.LibMatchMirror;
-                menuStrip.Items.Insert(iInsert++, mirrorMenuItem);
-                menuStrip.Items.Insert(iInsert++, toolStripSeparator61);
+            if (isProteomic)
+            {
+                if (control?.ControlType == SpectrumControlType.LibraryMatch)
+                {
+                    prositLibMatchItem.Checked = Settings.Default.Prosit;
+                    menuStrip.Items.Insert(iInsert++, prositLibMatchItem);
+                    mirrorMenuItem.Checked = Settings.Default.LibMatchMirror;
+                    menuStrip.Items.Insert(iInsert++, mirrorMenuItem);
+                    menuStrip.Items.Insert(iInsert++, toolStripSeparator61);
+                }
+            }
+            else
+            {
+                prositLibMatchItem.Checked = _graphSpectrumSettings.Prosit = false;
```

Comment:
It's weird that you're changing the value of "_graphSpectrumSettings.Prosit" (which changes Settings.Default.Prosit).

You would not think that the act of bringing up the right-click menu would change settings. (And also, that if this setting does need to be false, it's not going to be the correct behavior if you wait until the context menu is shown to change it).

Normally, you would make it so that would check whether what it is currently displaying is proteomic, and AND that with Settings.Default.Prosit.

Reply:
Yeah, I take your point. Starting over...